### PR TITLE
Tidy up purchase_helper a bit

### PR DESCRIPTION
**Pull request by Frost**
- Break long lines in PurchaseHelper and purchase form
- Increase test coverage
- Move some logic from purchase form into Purchase

### CODE REVIEW

File: app/models/purchase.rb [hunk 1]
```diff
@@ -163,4 +163,11 @@ def self.pay_multiple!(params)
 
     purchases.map(&:id)
   end
+
+  def self.states_collection
+    workflow_spec.states.map{ |s|
```
Comment (Koronen) [inline app/models/purchase.rb]: Missing space before `{`.

File: app/models/purchase.rb [hunk 2]
```diff
@@ -32,7 +32,8 @@ def self.searchable_language
   scope :unpaid, -> { where(workflow_state: %w(new edited confirmed bookkept)) }
   scope :confirmed, -> { where(workflow_state: %w(confirmed edited)) }
   scope :keepable, -> { where(workflow_state: :paid) }
-  scope :accepted, -> { where(workflow_state: %w(confirmed bookkept paid finalized)) }
+  scope :accepted, -> {
+    where(workflow_state: %w(confirmed bookkept paid finalized)) }
```
Comment (Frost) [inline app/models/purchase.rb]: Waah, waah... And if I put it on one line, that line will be too long. Damned if I do, damned if I don't. :(

Comment (Koronen) [inline app/models/purchase.rb]: Move the closing `}` to the line below?

**Comment by Yopi** [comment]
LGTM!

**Pull request merged**